nixos/modules/services/networking/mptcp/mptcp_up_raw

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+# A script for setting up routing tables for MPTCP
+
+# can check with (ins)[root@client:~]# journalctl -b -u NetworkManager-dispatcher.service
+# Copy this script into /etc/network/if-up.d/
+
+set -ex
+
+STATUS=$2
+
+# TODO remove
+echo $PATH
+env > /tmp/if_up_env
+
+RT_TABLE=/etc/iproute2/rt_tables
+
+if [ "$DEVICE_IFACE" = "lo" ]; then
+
+	logger -p user.debug -t mptcp "if localhost or $MODE then abort "
+	exit 0
+fi
+
+if [ -z "$DEVICE_IFACE" ]; then
+	logger -p user.warn -t mptcp "invalid $DEVICE_IFACE"
+	exit 0
+fi
+
+
+if [ `grep "$DEVICE_IFACE" "$RT_TABLE" | wc -l` -eq 0 ]; then
+	logger -p user.info -t mptcp "Adding new routing table $DEVICE_IFACE"
+	NUM=$(wc -l < "$RT_TABLE")
+	echo "$NUM  $DEVICE_IFACE" >> "$RT_TABLE"
+fi
+
+if [ "$DHCP4_IP_ADDRESS" ]; then
+	SUBNET=`echo $IP4_ADDRESS_0 | cut -d \   -f 1 | cut -d / -f 2`
+	ip route add table "$DEVICE_IFACE" to "$DHCP4_NETWORK_NUMBER/$SUBNET" dev "$DEVICE_IFACE" scope link
+	ip route add table "$DEVICE_IFACE" default via $DHCP4_ROUTERS dev "$DEVICE_IFACE"
+	ip rule add from "$DHCP4_IP_ADDRESS" table "$DEVICE_IFACE"
+else
+	# PPP-interface
+	IPADDR=`echo $IP4_ADDRESS_0 | cut -d \   -f 1 | cut -d / -f 1`
```

Comment:
ipv6?

Reply:
what's IPv6 :D ? I haven't tested with IPv6 (haven't done much IPv6 in general). I can try but if someone wanna step up before or after the PR, I would be really happy.